velox/type/Filter.h

```diff
@@ -983,6 +1004,30 @@ class BigintValuesUsingHashTable final : public Filter {
     }
   }
 
+  bool testBloomFilter(
+      const AbstractBloomFilter& bloomFilter,
+      const velox::Type& type) const override {
+    // Limit checks to IN-list with upto 10 values.
+    if (values_.size() > kMaxBloomFilterChecks) {
+      return true;
+    }
+    // For IN-list, if any value matches, return true.
+    for (auto i = 0; i < values_.size() && i < 10; ++i) {
+      if (type.kind() == TypeKind::INTEGER) {
```

Comment:
Don't test the type inside of the loop. Extract the type check out and use two separate loops